docs/en/stack/ml/nlp/ml-nlp-text-emb-vector-search-example.asciidoc

```diff
@@ -0,0 +1,339 @@
+where the ingest processor stores the embeddings. The msmarco-MiniLM-L-12-v3 model produces 
+embeddings with 384 dimensions, the `dense_vector` field must be configured 
+with the same number of dimensions as specified by the `dims` option.
+
+[source,js]
+--------------------------------------------------
+PUT collection-with-embeddings
+{
+  "mappings": {
+    "properties": {
+      "text_embedding.predicted_value": {
+        "type": "dense_vector",
+        "dims": 384,
+        "index": true,
+        "similarity": "cosine"
+      },
+      "text": {
+        "type": "text"
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+Create the text embeddings by reindexing the data to the 
+`collection-with-embeddings` index through the {infer} pipeline. The {infer} 
+ingest processor inserts the embedding vector into each document.
+
+[source,js]
+--------------------------------------------------
+POST _reindex?wait_for_completion=false
+{
+  "source": {
+    "index": "collection"
+  },
+  "dest": {
+    "index": "collection-with-embeddings",
+    "pipeline": "text-embeddings"
+  }
+}
+--------------------------------------------------
+
+The API call returns a task ID that can be used to monitor the progress:
+
+[source,js]
+--------------------------------------------------
+GET _tasks/<task_id>
+--------------------------------------------------
+
+You can also open the model stat UI to follow the progress.
+
+[role="screenshot"]
+image::images/ml-nlp-text-emb-reindex.png[Model status UI]
+
+After the reindexing is finished, the documents in the new index contain the 
+{infer} results – the vector embeddings. 
+
+
+[discrete]
+[[ex-text-emb-vect-search]]
+== Vector similarity search
+
+To perform vector similarity search, you need to obtain the text embedding of a 
+text. This example uses the "How is the weather in Jamaica?" query as the input 
+text. The {ref}/infer-trained-model-deployment.html[_infer API] gives you the 
+embedding of this query as a dense vector:
+
+[source,js]
+--------------------------------------------------
+POST /_ml/trained_models/sentence-transformers__msmarco-minilm-l-12-v3/_infer
+{
+  "docs": {
+    "text_field": "How is the weather in Jamaica?"
+  }
+}
+--------------------------------------------------
+
+You can use the dense vector in the `collection-with-embeddings` index by doing 
+{ref}/knn-search.html[kNN search]:
+
+[source,js]
+--------------------------------------------------
+GET collection-with-embeddings/_search
+{
+  "knn": {
+    "field": "text_embedding.predicted_value",
+    "query_vector": [
+     0.39521875977516174,
+        -0.3263707458972931,
+        0.26809820532798767,
+        0.30127981305122375,
+     (...)
+    ],
+    "k": 10,
+    "num_candidates": 100
+  },
+  "_source": [
+    "id",
+    "text"
+  ]
+}
+--------------------------------------------------
+
+As a result, you receive the top 10 closest to the query documents from the 
```

Comment:
This feels like a word is missing:
> As a result, you receive the top 10 closest xxx to the query documents from the 

Is it documents? Passages? Strings?

Reply:
The expression "closest to the query" works as an adjective that refers to the noun "documents". I rephrased it to the following: "As a result, you receive the top 10 documents that are the closest in meaning to the query..."